src/onchain_reading.js

```diff
@@ -22,18 +25,35 @@ const getOpenOrdersPaginated = async function* (contract, pageSize) {
   }
 }
 
+/**
+ * Returns open orders in the orderbook.
+ * @param {BatchExchange} contract to query from
+ * @param {number} pageSize the number of items to fetch per page
+ * @param {number} blockNumber the block number to execute the query at
+ */
+const getOpenOrders = async (contract, pageSize, blockNumber) => {
+  let allOrders = []
+  for await (const page of getOpenOrdersPaginated(contract, pageSize, blockNumber)) {
+    allOrders = allOrders.concat(page)
+  }
+  return allOrders
+}
+
 /**
  * Returns all orders in the orderbook.
  * @param {BatchExchange} contract to query from
  * @param {number} pageSize the number of items to fetch per page
+ * @param {number} blockNumber the block number to execute the query at
  */
-const getOrdersPaginated = async (contract, pageSize) => {
+const getOrdersPaginated = async (contract, pageSize, blockNumber) => {
```

Comment:
Given this is a method that is used by dependent projects can we make the default still be latest? This might be the case already (looking at the exposed ts definition, but can we document it)?

Reply:
It will default to `undefined` which is the current behaviour (kind of like not passing it at all to the `call` method.

I will add a note in the doc comments.